src/helpers.cpp

```diff
@@ -246,6 +246,29 @@ namespace ada::helpers {
     if (url.query.has_value()) return;
     while (!url.path.empty() && url.path.back() == ' ') { url.path.resize(url.path.size()-1); }
   }
+
+  ada_really_inline size_t find_authority_delimiter(const bool& is_special, std::string_view view) noexcept {
```

Comment:
I recommend to pass is_special by value. It won't matter to performance, I think, but you do actually want the compiler to *copy* is_special (it only takes one word). The const reference idiom is more useful when copying is expensive (e.g., when there is a non-trivial constructor as in std::string).

Won't make a performance difference, I think...